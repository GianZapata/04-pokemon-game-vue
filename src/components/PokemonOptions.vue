<template>
  <div class="flex flex-col items-center justify-center">
    <button
      class="font-bold py-2 px-4 rounded w-96 my-2 transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110 bg-blue-500 hover:bg-blue-700 text-white capitalize"
      v-for="pokemon in pokemons"
      :key="pokemon.id"
      @click="checkAnswer(pokemon.id)"
    >
      {{ capitalize(pokemon.name) }}
    </button>
  </div>
</template>

<script setup lang="ts">
import { usePokemons } from "@/composables/usePokemons"
import capitalize from "@/helpers/capitalize"

const { pokemons, checkAnswer } = usePokemons()
</script>
