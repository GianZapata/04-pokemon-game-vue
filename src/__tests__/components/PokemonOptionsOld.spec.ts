import { describe, test, expect, beforeEach } from "vitest"
import { shallowMount, VueWrapper } from "@vue/test-utils"
import PokemonOptions from "@/components/PokemonOptionsOld.vue"
import { mockedPokemons } from "../mocks/pokemons.mock"
import type { ComponentPublicInstance } from "vue"

describe("PokemonOptions Component", () => {
  let wrapper: VueWrapper<ComponentPublicInstance>

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons: mockedPokemons,
      },
    })
  })

  test("Debe de hacer el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test("Debe de mostrar las 4 opciones correctamente", () => {
    const buttonTags = wrapper.findAll("button")

    expect(buttonTags.length).toBe(4)

    expect(buttonTags[0].text().toLowerCase()).toBe("bulbasaur")
    expect(buttonTags[1].text().toLowerCase()).toBe("ivysaur")
    expect(buttonTags[2].text().toLowerCase()).toBe("venusaur")
    expect(buttonTags[3].text().toLowerCase()).toBe("charmander")
  })

  test("Debe de emitir el evento 'onSelectPokemon' con sus respectivos argumentos al hacer click", () => {
    const [buttonBulbasaur, buttonIvysaur, buttonVenusaur, buttonCharmander] =
      wrapper.findAll("button")

    buttonBulbasaur.trigger("click")
    buttonIvysaur.trigger("click")
    buttonVenusaur.trigger("click")
    buttonCharmander.trigger("click")

    expect(wrapper.emitted("selectPokemon")?.length).toBe(4)
    expect(wrapper.emitted("selectPokemon")?.[0]).toEqual([1])
    expect(wrapper.emitted("selectPokemon")?.[1]).toEqual([2])
    expect(wrapper.emitted("selectPokemon")?.[2]).toEqual([3])
    expect(wrapper.emitted("selectPokemon")?.[3]).toEqual([4])
  })
})
