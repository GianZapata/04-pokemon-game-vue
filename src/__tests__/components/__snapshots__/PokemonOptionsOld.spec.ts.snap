// Vitest Snapshot v1

exports[`PokemonOptions Component > Debe de hacer el snapshot 1`] = `"<div class=\\"flex flex-col items-center justify-center\\"><button class=\\"font-bold py-2 px-4 rounded w-96 my-2 transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110 bg-blue-500 hover:bg-blue-700 text-white\\">Bulbasaur</button><button class=\\"font-bold py-2 px-4 rounded w-96 my-2 transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110 bg-blue-500 hover:bg-blue-700 text-white\\">Ivysaur</button><button class=\\"font-bold py-2 px-4 rounded w-96 my-2 transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110 bg-blue-500 hover:bg-blue-700 text-white\\">Venusaur</button><button class=\\"font-bold py-2 px-4 rounded w-96 my-2 transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110 bg-blue-500 hover:bg-blue-700 text-white\\">Charmander</button></div>"`;
